File: Week03/LeetCode-0220.cpp
class Solution {
public:
    bool containsNearbyAlmostDuplicate(vector<int>& nums, int k, int t) {
        /// Handle corner case
        if (nums.size() < 2) return false;
        
        /// O(NlogN) to copy
        vector<pair<int, int>> A;
        for (int i=0; i<nums.size(); i++)
            A.push_back({nums[i], i});

        // Sort: O(NlogN)
        std::sort(A.begin(), A.end());
        
        // Walk over A
        for (int i=0; i<A.size()-1; i++){
            for (int j=i+1; j<A.size(); j++){
                if (abs((int64_t)A[i].first - (int64_t)A[j].first) > t)
                    break;
                if (std::abs(A[i].second-A[j].second) <= k)
                    return true;
            } // end-for
        } // end-for
        
        return false;
    } // end-containsNearbyAlmostDuplicate
};

File: Week03/LeetCode-0220.py
class Solution:
    def containsNearbyAlmostDuplicate(self, nums: List[int], k: int, t: int) -> bool:
        # Handle corner case
        if len(nums) < 2: return False
        
        # O(N) to copy
        A = []
        for i in range(len(nums)): 
            A.append([nums[i], i])
                        
        # Sort: O(NlogN)
        A.sort()
        
        # Walk over A.
        for i in range(len(A)-1):
            for j in range(i+1, len(A)):
                if abs(A[i][0]-A[j][0]) > t:
                    break
                if abs(A[i][1]-A[j][1]) <= k:
                    return True        
        return False
